### ISSUE TITLE
problem when execute mvn package with marven

### ISSUE THREAD

**Issue by jabberimaher** [eclipse/californium#833]
hi,
please when i launch mvn package to test marven it display :+1: Building mon-appli 1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ mon-appli ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/maher/projet1/mon-appli/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ mon-appli ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 1 source file to /home/maher/projet1/mon-appli/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] Source option 5 is no longer supported. Use 6 or later.
[ERROR] Target option 1.5 is no longer supported. Use 1.6 or later.
[INFO] 2 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.503 s
[INFO] Finished at: 2018-12-22T09:19:47+01:00
[INFO] Final Memory: 9M/37M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project mon-appli: Compilation failure: Compilation failure: 
[ERROR] Source option 5 is no longer supported. Use 6 or later.
[ERROR] Target option 1.5 is no longer supported. Use 1.6 or later.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
how i can resolve this problem

**Comment by jabberimaher** [comment]
i use 1.5-snapshot the problem is here:
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] Source option 5 is no longer supported. Use 6 or later.
[ERROR] Target option 1.5 is no longer supported. Use 1.6 or later.
[INFO] 2 errors
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 1.503 s
[INFO] Finished at: 2018-12-22T09:19:47+01:00
[INFO] Final Memory: 9M/37M
best regards

**Comment by jabberimaher** [comment]
i'm sorry 
i can not add demo-apps / sc-dtls-example-server as server and demo-apps / sc-dtls-example-client as client.
I don't know if I add them by modifying only the pom.xml file or there are other steps, I installed californium 2.0.0.M12 then I have the following parent tree / src / org / eclipse / californium / app.java. ...
I need the pom.xml file needed for adding californium and demo-apps and the other steps if they exist.
thank you

**Comment by boaks** [comment]
Californium 2.0.0-M12 already contains the demo-apps! You don't need to add them.
At the end of the log from your maven build, there should be a Reactor Summary, which contains all the build module. 

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Californium (Cf) Parent ............................ SUCCESS [  1.173 s]
[INFO] Californium (Cf) Demo Certificates ................. SUCCESS [  1.395 s]
[INFO] element-connector .................................. SUCCESS [ 13.982 s]
[INFO] Scandium (Sc) Core ................................. SUCCESS [05:06 min]
[INFO] element-connector-tcp .............................. SUCCESS [ 31.808 s]
[INFO] Californium (Cf) Core .............................. SUCCESS [04:18 min]
[INFO] Californium (Cf) Demo Applications ................. SUCCESS [  0.021 s]
[INFO] Cf-NAT ............................................. SUCCESS [  3.977 s]
[INFO] Californium (Cf) Integration Tests ................. SUCCESS [ 23.429 s]
[INFO] Californium (Cf) Proxy ............................. SUCCESS [  2.074 s]
[INFO] Californium (Cf) OSGi .............................. SUCCESS [  2.212 s]
[INFO] Sc-DTLS-Example-Server ............................. SUCCESS [  1.648 s]
[INFO] Sc-DTLS-Example-Client ............................. SUCCESS [  1.348 s]
[INFO] Cf-PlugtestClient .................................. SUCCESS [  3.391 s]
[INFO] Cf-PlugtestChecker ................................. SUCCESS [  3.869 s]
[INFO] Cf-PlugtestServer .................................. SUCCESS [  2.543 s]
[INFO] Cf-ExtendedPlugtestClient .......................... SUCCESS [  3.887 s]
[INFO] Cf-ExtendedPlugtestServer .......................... SUCCESS [  3.886 s]
[INFO] Cf-HelloWorldClient ................................ SUCCESS [  0.944 s]
[INFO] Cf-HelloWorldServer ................................ SUCCESS [  2.543 s]
[INFO] Cf-SimpleFileServer ................................ SUCCESS [  0.928 s]
[INFO] Cf-CoCoAClient ..................................... SUCCESS [  1.211 s]
[INFO] Cf-SecureServer .................................... SUCCESS [  1.151 s]
[INFO] Cf-ExampleCrossProxy ............................... SUCCESS [  4.425 s]
[INFO] Cf-BenchmarkServer ................................. SUCCESS [  2.604 s]
[INFO] Cf-ObserveBenchmarkClient .......................... SUCCESS [  0.962 s]
[INFO] oscore-cf .......................................... SUCCESS [  2.439 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 11:26 min
[INFO] Finished at: 2018-12-28T10:23:30+01:00
[INFO] Final Memory: 29M/139M
[INFO] ------------------------------------------------------------------------
```

Do you get a similar summary? If so, https://github.com/eclipse/californium/issues/787#issuecomment-449651404 should contain the instructions to execute the examples.

> If it builds, you may use `demo-apps/sc-dtls-example-server` as server and `demo-apps/sc-dtls-example-client` as client. To do so, open a terminal in the californium/demo-apps/run folder and execute `java -jar sc-dtls-example-server-2.0.0-SNAPSHOT.jar` and afterwards from an other terminal in that folder `java -jar sc-dtls-example-client-2.0.0-SNAPSHOT.jar`.
> The client should the print
> 
> > Create 1 DTLS example clients, expect to send 100 messages overall.
> > 1 DTLS example clients started.
> > 1 DTLS example clients finished.
> > 100 messages received, 100 expected
> > 100 messages in 338 ms
> > 295 messages per s
> 
> Or something similar.
> 
> > how i realize this work and can i implement others authentication algorithms?
> 
> This depends of the scope of your "authentication algorithms" and your work.
>  
> If your intention is to "authenticate" the other peer, californium offers ciphersuites with PSK (PreSharedKey), Raw Public Key, x509v3 Certificate based ciphersuites.
> 
> TLS_PSK_WITH_AES_128_CBC_SHA256
> TLS_ECDHE_PSK_WITH_AES_128_CBC_SHA256
> TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256
> TLS_PSK_WITH_AES_128_CCM_8
> TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8
> 
> So, if you want to compare them, you may use demo-apps/cf-secure project.
> Also open two terminals in californium/demo-apps/run and start the server with
> 
> > java -jar cf-secure-2.0.0-SNAPSHOT.jar
> 
> and the client with
> 
> > java -cp cf-secure-2.0.0-SNAPSHOT.jar org.eclipse.californium.examples.SecureClient PSK
> 
> At the end of the console output, there should be something as:
>  
> ```
> ==[ CoAP Response ]============================================
> MID    : 27727
> Token  : 746DFCEF9FACAC46
> Type   : ACK
> Status : 2.05
> Options: {"Content-Format":"text/plain"}
> RTT    : 129 ms
> Payload: 14 Bytes
> ---------------------------------------------------------------
> hello security
> ===============================================================
> ```
> 
> You may replace PSK with RPK or X590.
> 
> If you want to compare with other ciphersuites, e.g. some with MD5, I would recommend to use some TLS (openssl) libraries, which offers that. To extend californium with old MD5 is not planned at all and will be extremely hard to do.
> 
> I' still not understand, why you want to use java to measure the energy consumption.  
> 

So: do you have a "demo-apps" folder? Does it contain a "run" folder? Are the three jars mention above in your reactor summary and in the run folder? Are you able to open a command terminal in that folder (alternatively change the cd of your command terminal to the "demo-apps/run" folder)?

**Comment by jabberimaher** [comment]
this isthe file pom.xml

**Comment by jabberimaher** [comment]
1-root@maher-Lenovo-G50-80:/home/maher/p1# mvn archetype:generate -DarchetypeArtifactId=maven-archetype-quickstart -DarchetypeVersion=1.1
2-[WARNING] No archetype found in remote catalog. Defaulting to internal catalog
Define value for property 'groupId': org.eclipse.californium
Define value for property 'artifactId': POM
Define value for property 'version' 1.0-SNAPSHOT: : 2.0-sNAPSHOT
Define value for property 'package' org.eclipse.californium: : 
3- it dispaly:
[INFO] ----------------------------------------------------------------------------
[INFO] Using following parameters for creating project from Old (1.x) Archetype: maven-archetype-quickstart:1.1
[INFO] ----------------------------------------------------------------------------
[INFO] Parameter: basedir, Value: /home/maher/p1
[INFO] Parameter: package, Value: org.eclipse.californium
[INFO] Parameter: groupId, Value: org.eclipse.californium
[INFO] Parameter: artifactId, Value: POM
[INFO] Parameter: packageName, Value: org.eclipse.californium
[INFO] Parameter: version, Value: 2.0-sNAPSHOT
[INFO] project created from Old (1.x) Archetype in dir: /home/maher/p1/POM
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:14 min
[INFO] Finished at: 2018-12-28T11:03:16+01:00
[INFO] Final Memory: 15M/57M
[INFO] -----------------------------------
4-root@maher-Lenovo-G50-80:/home/maher/p1# cd POM
root@maher-Lenovo-G50-80:/home/maher/p1/POM# mvn package
5-this is my pom.xml:
"--------------------<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>org.eclipse.californium</groupId>
  <artifactId>POM</artifactId>
  <version>2.0-sNAPSHOT</version>
  <packaging>jar</packaging>

  <name>POM</name>
  <url>http://maven.apache.org</url>

  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
  </properties>
<build>
    
    <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-compiler-plugin</artifactId>
        <version>3.8.0</version>
        <configuration>
          <source>1.8</source>
          <target>1.8</target>
        </configuration>
      </plugin>
    </plugins>
    
  </build>

  <dependencies>

<dependency>
    <groupId>org.eclipse.californium</groupId>
    <artifactId>californium-core</artifactId>
    <version>2.0.0-M12</version>
</dependency>
<dependency>
    <groupId>org.eclipse.californium</groupId>
    <artifactId>scandium</artifactId>
    <version>2.0.0-M12</version>
</dependency>

    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>3.8.1</version>
      <scope>test</scope>
    </dependency>
</dependencies>-->
</project>
6-[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building POM 2.0-sNAPSHOT
[INFO] ------------------------------------------------------------------------
Downloading from central: https://repo.maven.apache.org/maven2/org/eclipse/californium/scandium/2.0.0-M12/scandium-2.0.0-M12.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/eclipse/californium/scandium/2.0.0-M12/scandium-2.0.0-M12.pom (4.8 kB at 4.7 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/eclipse/californium/scandium/2.0.0-M12/scandium-2.0.0-M12.jar
Downloaded from central: https://repo.maven.apache.org/maven2/org/eclipse/californium/scandium/2.0.0-M12/scandium-2.0.0-M12.jar (315 kB at 465 kB/s)
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ POM ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/maher/p1/POM/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ POM ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ POM ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/maher/p1/POM/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ POM ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ POM ---
[INFO] Surefire report directory: /home/maher/p1/POM/target/surefire-reports

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.eclipse.californium.AppTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.025 sec

Results :

Tests run: 1, Failures: 0, Errors: 0, Skipped: 0

[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ POM ---
[INFO] Building jar: /home/maher/p1/POM/target/POM-2.0-sNAPSHOT.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
-----------------------
it build the folder californium but i's empty......!!!!!

**Comment by boaks** [comment]
No glue, what your doing! Seems to be totally mixed up! 

Did you download californium? If yes, which link did you use?
Or did you clone the repo? If yes, which link did you use as remote-repository?

If you open the web-page https://github.com/eclipse/californium you see the instructions to build californium (https://github.com/eclipse/californium#build-using-maven).
I don't know, why you use `mvn archetype:generate -DarchetypeArtifactId=maven-archetype-quickstart -DarchetypeVersion=1.1`. 

So:
- download or clone californium
- open a command-terminal in the californium parent folder
- build it as documented under the link above.

Doing so should print after some lines `Building Californium (Cf) Parent 2.0.0-SNAPSHOT` and **NOT** `2.0-sNAPSHOT`.

**Comment by jabberimaher** [comment]
californium is installed with success
i launch in the first terminal root@maher-Lenovo-G50-80:/home/maher/californium/demo-apps/run# java -jar sc-dtls-example-server-1.1.0-SNAPSHOT.jar
2018-12-28 17:00:29:   1 CONFIG [InMemoryConnectionStore]: Created new InMemoryConnectionStore [capacity: 150,000, connection expiration threshold: 1,800s] - (org.eclipse.californium.scandium.dtls.InMemoryConnectionStore.java:124) <init>() in thread main
2018-12-28 17:00:29:   1 CONFIG [DTLSConnector]: Cannot determine MTU of network interface, using minimum MTU [1280] of IPv6 instead - (org.eclipse.californium.scandium.DTLSConnector.java:350) start() in thread main
2018-12-28 17:00:29:   1 INFO [DTLSConnector]: DTLS connector listening on [0.0.0.0/0.0.0.0:5684] with MTU [1,280] using (inbound) datagram buffer size [16,474 bytes] - (org.eclipse.californium.scandium.DTLSConnector.java:374) start() in thread main
2018-12-28 17:00:29:  12 CONFIG [DTLSConnector$Worker]: Starting worker thread [DTLS-Receiver-0.0.0.0/0.0.0.0:5684] - (org.eclipse.californium.scandium.DTLSConnector$Worker.java:1662) run() in thread DTLS-Receiver-0.0.0.0/0.0.0.0:5684
root@maher-Lenovo-G50-80:/home/maher/californium/demo-apps/run# 
in a second terminal i launch root@maher-Lenovo-G50-80:/home/maher/californium/demo-apps/run# java -jar sc-dtls-example-client-1.1.0-SNAPSHOT.jar

2018-12-28 17:01:42:   1 CONFIG [InMemoryConnectionStore]: Created new InMemoryConnectionStore [capacity: 150,000, connection expiration threshold: 1,800s] - (org.eclipse.californium.scandium.dtls.InMemoryConnectionStore.java:124) <init>() in thread main
2018-12-28 17:01:42:   1 CONFIG [DTLSConnector]: Cannot determine MTU of network interface, using minimum MTU [1280] of IPv6 instead - (org.eclipse.californium.scandium.DTLSConnector.java:350) start() in thread main
2018-12-28 17:01:42:   1 INFO [DTLSConnector]: DTLS connector listening on [0.0.0.0/0.0.0.0:58214] with MTU [1,280] using (inbound) datagram buffer size [16,474 bytes] - (org.eclipse.californium.scandium.DTLSConnector.java:374) start() in thread main
2018-12-28 17:01:42:  12 CONFIG [DTLSConnector$Worker]: Starting worker thread [DTLS-Receiver-0.0.0.0/0.0.0.0:58214] - (org.eclipse.californium.scandium.DTLSConnector$Worker.java:1662) run() in thread DTLS-Receiver-0.0.0.0/0.0.0.0:58214
2018-12-28 17:01:42:  13 FINE [Connection]: Handshake with [localhost/127.0.0.1:5684] has been started - (org.eclipse.californium.scandium.dtls.Connection.java:239) handshakeStarted() in thread pool-1-thread-1
2018-12-28 17:01:42:  13 FINE [ECDHECryptography$SupportedGroup]: Group [brainpoolP256r1] is not supported by JRE - (org.eclipse.californium.scandium.dtls.cipher.ECDHECryptography$SupportedGroup.java:436) <init>() in thread pool-1-thread-1
2018-12-28 17:01:42:  13 FINE [ECDHECryptography$SupportedGroup]: Group [brainpoolP384r1] is not supported by JRE - (org.eclipse.californium.scandium.dtls.cipher.ECDHECryptography$SupportedGroup.java:436) <init>() in thread pool-1-thread-1
2018-12-28 17:01:42:  13 FINE [ECDHECryptography$SupportedGroup]: Group [brainpoolP512r1] is not supported by JRE - (org.eclipse.californium.scandium.dtls.cipher.ECDHECryptography$SupportedGroup.java:436) <init>() in thread pool-1-thread-1
2018-12-28 17:01:43:  15 FINE [DTLSConnector]: Re-transmitting flight for [localhost/127.0.0.1:5684], [3] retransmissions left - (org.eclipse.californium.scandium.DTLSConnector.java:1485) handleTimeout() in thread pool-1-thread-2
2018-12-28 17:01:45:  16 FINE [DTLSConnector]: Re-transmitting flight for [localhost/127.0.0.1:5684], [2] retransmissions left - (org.eclipse.californium.scandium.DTLSConnector.java:1485) handleTimeout() in thread pool-1-thread-3
2018-12-28 17:01:49:  18 FINE [DTLSConnector]: Re-transmitting flight for [localhost/127.0.0.1:5684], [1] retransmissions left - (org.eclipse.californium.scandium.DTLSConnector.java:1485) handleTimeout() in thread pool-1-thread-4
2018-12-28 17:01:57:  19 FINE [DTLSConnector]: Re-transmitting flight for [localhost/127.0.0.1:5684], [0] retransmissions left - (org.eclipse.californium.scandium.DTLSConnector.java:1485) handleTimeout() in thread pool-1-thread-5
2018-12-28 17:02:13:  20 FINE [DTLSConnector]: Flight for [localhost/127.0.0.1:5684] has reached maximum no. [4] of retransmissions, discarding ... - (org.eclipse.californium.scandium.DTLSConnector.java:1502) handleTimeout() in thread pool-1-thread-6
it still running now and i wait for the result.........................

**Comment by boaks** [comment]
> 1.1.0-SNAPSHOT

Your on the master branch. But 2.0.x (2.0.0-SNAPSHOT or 2.0.0-M12) was recommended.

>it still running now and i wait for the result

I still wait on the answers to my questions :-)

https://github.com/eclipse/californium/issues/833#issuecomment-450354325

**Comment by jabberimaher** [comment]
question :why you want to use java to measure the energy consumption.?????:-)
professor , i need to compare between dtls handshake and MDA, then i need a simulator who performs this task i tried contiki, but after three months i did nothing.
I was advised to use another one and this is californium.
.....................................
i think there is problem when running example dtls client, almost 3 hours of execution of the order and it does not show anything, I think it is blocked..no?.

.

**Comment by boaks** [comment]
The link was to my comment above, the questions are:

> Did you download californium? If yes, which link did you use?
> Or did you clone the repo? If yes, which link did you use as remote-repository?

That's important to know, what you did, because, you obvious using the wrong stuff (1.1.0 instead of the recommended 2.0.0).

> i think there is problem when running example dtls client, almost 3 hours of execution of the order and it does not show anything, I think it is blocked..no?.

Sure, there is a problem running a **DIFFERENT VERSION** of the examples than recommended!
In your case, the server accidentally terminated (caused by a missing fix, which is already applied to 2.0.x), therefore the client fails also.

But again, if you do things different, don't complain, if it's not working.

So answer the two questions above, then I know, what you have and how to help you, to get the right stuff.

**Comment by jabberimaher** [comment]
link download californium: https://github.com/eclipse/californium#build-using-maven

**Comment by jabberimaher** [comment]
i downloaded californium with browser, i didn't clone it

**Comment by boaks** [comment]
If you select the right branch 2.0.x, the download link is 

https://github.com/eclipse/californium/archive/2.0.x.zip

**Comment by jabberimaher** [comment]
hello professor, I apologize for the long break, but I was sick and I can not move my legs.
i download californium from the recommended link and using the described steps but at the end of installation it displays this error:
[ERROR] /home/maher/californium/californium-core/src/main/java/org/eclipse/californium/core/network/config/NetworkConfig.java:406: Undefined reference: void AutoCloseable.close()
[ERROR] /home/maher/californium/californium-core/src/main/java/org/eclipse/californium/core/network/config/NetworkConfig.java:406: Undefined reference: void Throwable.addSuppressed(Throwable)
[ERROR] /home/maher/californium/californium-core/src/main/java/org/eclipse/californium/core/network/config/NetworkConfig.java:406: Undefined reference: void AutoCloseable.close()
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Californium (Cf) Parent ............................ SUCCESS [  3.175 s]
[INFO] Californium (Cf) Demo Certificates ................. SUCCESS [  2.352 s]
[INFO] element-connector .................................. SUCCESS [ 15.812 s]
[INFO] Scandium (Sc) Core ................................. SUCCESS [01:07 min]
[INFO] element-connector-tcp .............................. SUCCESS [ 29.525 s]
[INFO] Californium (Cf) Core .............................. FAILURE [  1.577 s]
[INFO] Californium (Cf) Demo Applications ................. SKIPPED
[INFO] Cf-NAT ............................................. SKIPPED
[INFO] Californium (Cf) Integration Tests ................. SKIPPED
[INFO] Californium (Cf) Proxy ............................. SKIPPED
[INFO] Californium (Cf) OSGi .............................. SKIPPED
[INFO] Sc-DTLS-Example-Server ............................. SKIPPED
[INFO] Sc-DTLS-Example-Client ............................. SKIPPED
[INFO] Cf-PlugtestClient .................................. SKIPPED
[INFO] Cf-PlugtestChecker ................................. SKIPPED
[INFO] Cf-PlugtestServer .................................. SKIPPED
[INFO] Cf-ExtendedPlugtestClient .......................... SKIPPED
[INFO] Cf-ExtendedPlugtestServer .......................... SKIPPED
[INFO] Cf-HelloWorldClient ................................ SKIPPED
[INFO] Cf-HelloWorldServer ................................ SKIPPED
[INFO] Cf-SimpleFileServer ................................ SKIPPED
[INFO] Cf-CoCoAClient ..................................... SKIPPED
[INFO] Cf-SecureServer .................................... SKIPPED
[INFO] Cf-ExampleCrossProxy ............................... SKIPPED
[INFO] Cf-BenchmarkServer ................................. SKIPPED
[INFO] Cf-ObserveBenchmarkClient .......................... SKIPPED
[INFO] oscore-cf .......................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:06 min
[INFO] Finished at: 2019-01-06T12:30:17+01:00
[INFO] Final Memory: 34M/120M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:animal-sniffer-maven-plugin:1.16:check (default) on project californium-core: Signature errors found. Verify them and ignore them with the proper annotation if needed. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :californium-core
root@maher-Lenovo-G50-80:/home/maher/californium#
best regards

**Comment by boaks** [comment]
Do you use Java 1.6?

Check may be done with `java -version`  of `javac -version`.

**Comment by boaks** [comment]
If you use java 1.6, please try to update at least to 1.7.  
(May be it's easier to update to 1.8.)

**Comment by boaks** [comment]
If possible for you, I would like to cleanup the issue #787 from this "build discussion".
Would this be OK for you, if I delete your and my comments there?

**Comment by jabberimaher** [comment]
root@maher-Lenovo-G50-80:/home/maher/californium# javac -version
javac 1.8.0_191
root@maher-Lenovo-G50-80:/home/maher/californium# java -version
openjdk version "10.0.2" 2018-07-17
OpenJDK Runtime Environment (build 10.0.2+13-Ubuntu-1ubuntu0.18.04.4)
OpenJDK 64-Bit Server VM (build 10.0.2+13-Ubuntu-1ubuntu0.18.04.4, mixed mode)
root@maher-Lenovo-G50-80:/home/maher/californium#

**Comment by jabberimaher** [comment]
ok yes you can delete from 787

**Comment by boaks** [comment]
Seems you use different version for the compiler (javac 1.8) and runtime (java 10.0).
If possible, please use also the 1.8 for the runtime. 

Sometimes this is caused by using different "jdk" and "jre". To check that, use

`sudo apt list --installed openjdk*`

If that reports different versions, please uninstall the openjdk-10-jre (or what ever version is reported for the jre) and install  openjdk-8-jre.

**Comment by jabberimaher** [comment]
root@maher-Lenovo-G50-80:/home/maher/californium# javac -version
javac 1.8.0_191
root@maher-Lenovo-G50-80:/home/maher/californium# javac -version
javac 1.8.0_191
root@maher-Lenovo-G50-80:/home/maher/californium# java -version
openjdk version "1.8.0_191"
OpenJDK Runtime Environment (build 1.8.0_191-8u191-b12-0ubuntu0.18.04.1-b12)
OpenJDK 64-Bit Server VM (build 25.191-b12, mixed mode)
root@maher-Lenovo-G50-80:/home/maher/californium# mvn clean install
The JAVA_HOME environment variable is not defined correctly
This environment variable is needed to run this program
NB: JAVA_HOME should point to a JDK not a JRE

**Comment by boaks** [comment]
OK, now jdk and jre are both 1.8. Great!

> root@maher-Lenovo-G50-80:/home/maher/californium# mvn clean install
> The JAVA_HOME environment variable is not defined correctly
> This environment variable is needed to run this program
> NB: JAVA_HOME should point to a JDK not a JRE

That are general java and maven installation issues. Please use a internet search machine to solve it. Java installation is sure no californium specific issue.

**Comment by jabberimaher** [comment]
thanks a lot professor,
i resolve the problem by adding to  /etc/environment the line:
export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64.
i try dtls server/client and cf-secure, it's ok.
---------------------------------------------------------------------------------------------------------------------------------------
now if i want to create scenario wsn : a node sensor would  to authenticate with a cloud server and i measure the consumption energy of the sensor and the needed time of authentication.
i need to create new isuue??

**Comment by boaks** [comment]
> i try dtls server/client and cf-secure, it's ok.

Congratulation!

> now if i want to create scenario wsn : a node sensor would to authenticate with a cloud server and i measure the consumption energy of the sensor and the needed time of authentication.
> i need to create new isuue??

I'm not sure, what you expect as help. If you can execute the cf-secure demo, why are you not able to do these measurements? "Time" should be easy and "energy" depends more on your client. So both seems to be not really related to californium.
But sure, if you want, you may open a new issue and ask other californium users to share their experience with you. I'm not sure, if someone has already that special experience.

**Comment by boaks** [comment]
If it builds now, can we close this issue?
